core/arch/arm/sm/sm.c

```diff
@@ -34,12 +35,8 @@ bool sm_from_nsec(struct sm_ctx *ctx)
 	if (sm_platform_handler(ctx) == SM_HANDLER_SMC_HANDLED)
 		return false;
 
-#ifdef CFG_PSCI_ARM32
-	if (OPTEE_SMC_OWNER_NUM(*nsec_r0) == OPTEE_SMC_OWNER_STANDARD) {
-		smc_std_handler((struct thread_smc_args *)nsec_r0, &ctx->nsec);
-		return false;	/* Return to non secure state */
-	}
-#endif
+	if (smc_std_handler(thread_args, &ctx->nsec) == SM_HANDLER_SMC_HANDLED)
```

Comment:
While this looks lean and clean it's actually less efficient than the code that it replaces in both configurations (with and without PSCI support) for all requests except PSCI requests which I guess is on par.

Since this is a critical path I think it makes sense to take every instruction into account.

Reply:
The `OPTEE_SMC_OWNER_NUM(*nsec_r0) == OPTEE_SMC_OWNER_STANDARD` test was also deliberate in order to avoid a non-local function call in the critical path.